spec/services/authentication/authenticator_spec.rb

```diff
@@ -273,6 +298,13 @@
         end.not_to change(Identity, :count)
       end
 
+      it "does not record an identity creation metric" do
+        allow(DatadogStatsClient).to receive(:increment)
+        service.call
+
+        expect(DatadogStatsClient).not_to have_received(:increment)
+      end
```

Comment:
Hm, what if an existing user created a new identity? I'm thinking of if someone auths with Twitter, and then later auths with GitHub...

Reply:
cc @Zhao-Andy here as I'm not sure how that scenario works if the user has two different emails. 

If they have the same email (as I do for both twitter and github) they will end up in 

https://github.com/thepracticaldev/dev.to/blob/8340051649b38cc35ea6ec18405bc975131e0aba/app/services/authentication/authenticator.rb#L69

Reply:
If someone has two different emails, we assume it's a new user being created, so a new user (and identity) will be created. That's where we get issues with users having two accounts and then later ask for them to be merged.

I think we _have_ to assume that for security reasons.

Reply:
Great write up @Zhao-Andy, thanks!

Reply:
@vaidehijoshi this is def only tracking identity creation rather than user creation. We will then alert under a given threshold for each provider. I choose to do it this way since we don't always create new users when authenticating like this. We could add tracking for user creation and other model creations as well in the future but here I thought focusing on identity was the way to go.